src/cltk/sentence/processes.py

```diff
@@ -0,0 +1,83 @@
+"""Module for sentence tokenizers.
+"""
+
+
+from copy import deepcopy
+from dataclasses import dataclass
+
+from boltons.cacheutils import cachedproperty
+
+from cltk.core import CLTKException
+from cltk.core.data_types import Doc, Process
+from cltk.sentence.non import OldNorseRegexSentenceTokenizer
+from cltk.sentence.sentence import SentenceTokenizer
+
+
+__author__ = ["Clément Besnier <clem@clementbesnier.fr>"]
+
+
+@dataclass
+class SentenceTokenizationProcess(Process):
+    """To be inherited for each language's tokenization declarations.
+
+    Example: ``SentenceTokenizationProcess`` -> ``OldNorseTokenizationProcess``
+
+    >>> from cltk.tokenizers.processes import TokenizationProcess
+    >>> from cltk.core.data_types import Process
+    >>> issubclass(SentenceTokenizationProcess, Process)
+    True
+    >>> tok = SentenceTokenizationProcess()
```

Comment:
Could we add a doctest that verifies the index behavior that occurs in the run() function?

Reply:
I don't know how to do it simply. I would need to make a class that inherits of `SentenceTokenizationProcess`, then we can compare `assert output_doc.words[-1].index_sentence == len(output_doc.sentences)`.

Reply:
It would be a nice-to-have, maybe somewhere else.